src/bosh-director/lib/bosh/director/deployment_plan/stages/create_network.rb

```diff
@@ -162,9 +162,10 @@ def populate_subnet_properties(subnet, db_subnet)
           subnet.restricted_ips.add(network_id.to_i)
           subnet.restricted_ips.add(broadcast.to_i)
           each_ip(JSON.parse(db_subnet.reserved)) do |ip|
-            unless subnet.range.contains?(ip)
+            ip_obj = ip_to_netaddr(ip)
+            unless subnet.range.contains(ip_obj)
               raise NetworkReservedIpOutOfRange,
-                    "Reserved IP '#{format_ip(ip)}' is out of " \
+                    "Reserved IP '#{format_ip(ip_obj)}' is out of " \
```

Comment:
I would use the raw ip from json here. Otherwise an exception could raise for. `{"ip" : {...}}` and you don't get the expected exception.
`"Reserved IP '#{ip}' is out of "`

Reply:
`each_ip` iterates over the json and supplies the integer representation of each ip. To have a human-readable error message the ip has to be converted back to its string representation. Therefore, either `format_ip(ip)` or `CIDRIP.parse(ip).to_s` are required.